app/controllers/api/v1/threat_levels_controller.rb

```diff
@@ -17,44 +17,83 @@ def set_limit
         @limit = @limit > 0 ? @limit : 5
       end
 
-      def threat_levels_data
-        {threat_levels: threat_levels_for_profiles}
-      end
-
       def threat_levels_for_profiles
-        Profile.all.collect { |p| threat_levels_for_profile(p) }
+        @assessments_data = {}
+        @threat_assessments = []
+        threat_assessment_ids = ThreatAssessment.find_by_sql(sql).collect {|ta| ta.id}
+        if threat_assessment_ids.present?
+          results_data = build_results_data(threat_assessment_ids)
+          fill_threat_assessments_data(results_data)
+        end
+        @assessments_data.values
       end
 
-      def threat_levels_for_profile(profile)
-        {
-          profile_id: profile.id,
-          threat_assessments: threat_assessments_for_profile(profile),
-          threat_level_assessment_runs: ThreatLevel::AssessmentRunsDataBuilderService.get(profile)
-        }
-      end
+      def fill_threat_assessments_data(results_data)
+        @threat_assessments.uniq!
+        @threat_assessments.each do |ta|
+          profile = ta.profiles.first
```

Comment:
Good question. We've this conversation when I started working on _porównywaczka_ years (?) ago. We agreed to have such an assumption that assessment is associated with one profile despite model enables N-M associations. _Porównywaczka_  always creates an assessment associated with one profile. All assessments in prod DB have at most one profile.
Although you are right there is a potential bug in the code and I'll change it if you insist.